spec/routes_utils_spec.js

```diff
@@ -870,71 +869,137 @@ describe('Utils: ', function() {
+      expect(index).toEqual(0);
     });
 
-    it('should update the existing investment and dividend if revoke', function() {
-      var existingInvestment = { _id: '1', user: 'Matt', userId: '123', amount: 10, percentage: 0.1, dividend: 10 };
-      var fromUser = { reps: 100, portfolio: [{ category: 'Coding', investments: [existingInvestment] }] };
-      amount = -2;
-      toUserReps = 998;
-      toUser = { id: '123', name: 'Matt' };
-      var id = '1';
-      var err = utils.updateTransactionFromUser(fromUser, toUser, category, amount, toUserReps, id);
-
-      expect(fromUser.portfolio[0].investments.length).toEqual(1);
-      expect(fromUser.portfolio[0].investments[0].amount).toEqual(8);
-      expect(fromUser.portfolio[0].investments[0].percentage).toEqual(0.08);
-      expect(fromUser.portfolio[0].investments[0].dividend).toEqual(7.98);
-      expect(fromUser.reps).toEqual(102);
-      expect(err).toEqual(null);
+    it('adds portfolio index if not found', function() {
+      var updatedFromUser = { portfolio: [{ category: 'Coding', investments: [] }, { category: 'Ballet', investments: [] }] }
+      fromUser = { portfolio: [{ category: 'Coding', investments: [] }] };
+      category = { name: 'Ballet', investors: 0 };
+      amount = 1;
+      toUserReps = 10;
+      var index = utils.getTransactionPortfolioIndex(amount, fromUser, toUser, toUserReps, null, category);
+      expect(index).toEqual(1);
+      expect(category.investors).toEqual(1);
     });
+  });
 
-    it('should remove the investment if it is entirely revoked', function() {
-      var investment = { _id: '1', user: "Matt", userId: "123", amount: 10, dividend: 1, percentage: 10 };
-      var investment2 = { _id: '2', user: "Bob", userId: "456", amount: 10, dividend: 1, percentage: 10 };
-      var investment3 = { _id: '3', user: "Matt", userId: "123", amount: 10, dividend: 1, percentage: 10 };
-      var fromUser = { reps: 100, portfolio: [{ category: "Coding", roi: { value: 0, length: 0 }, investments: [investment, investment2, investment3] }] };
-      amount = -10;
-      toUserReps = 990;
-      toUser = { id: '123', name: 'Matt' };
-      var id = '1';
-      spyOn(utils, 'removeInvestorFromExpertOnRevoke').andReturn();
-
-      var err = utils.updateTransactionFromUser(fromUser, toUser, category, amount, toUserReps, id);
-      expect(fromUser.portfolio[0].investments.length).toEqual(2);
-      expect(fromUser.portfolio[0].investments[0]._id).toEqual('2');
-      expect(fromUser.reps).toEqual(110);
-      expect(err).toEqual(null);
-    });
+  describe('addTransactionInvesment', function() {
+    var fromUser, category, toUser, amount, toUserReps;
+    it('should return an error if revoking from a non-existent investment', function() {
+      fromUser = { portfolio: [{ category: 'Coding', investments: [] }] };
+      category = 'Coding';
+      amount = -1;
```

Comment:
Is there any reason you make the amount equal to -1? It makes the test more complicated, because it's not clear whether the error will be thrown due to the amount value.